483e26259ba0530a42be722976df63a489a0e6d9..a6557f640a806096c618c84b80400120e6c419df
50fdd5e2b5c6347ccffb19efe31f271feca92de0..ce5477ef02688e4f535c6916c26193743ab38f40
7ab14aba1a1852724ef1aaa408c8bac6805919fb..8e4f7ddcef4a9b2913569e96431fe53c1fb20623
91e3ce8c5d1688cc71943d65dbf9604274b748b2..

From 6db51e779e0a40a241ddf3d6cc8eab42db495eec Mon Sep 17 00:00:00 2001
From: Robert Walker <bob.mt.wya@gmail.com>
Date: Mon, 27 Feb 2023 12:23:27 +0000
Subject: [PATCH] mscoree: support wine-mono downgrade

Fixes installation of an earlier version of wine-mono, to a WINEPREFIX,
if the current Wine version is downgraded.
The stock behaviour of Wine is to leave, newer versions of wine-mono,
installed in a WINEPREFIX. This will break any applications that require
wine-mono support.
---

 dlls/mscoree/mscoree_main.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)


diff --git a/dlls/mscoree/mscoree_main.c b/dlls/mscoree/mscoree_main.c
index 40577de7148..d395f816667 100644
--- a/dlls/mscoree/mscoree_main.c
+++ b/dlls/mscoree/mscoree_main.c
@@ -756,5 +756,5 @@ static BOOL get_support_msi(LPCWSTR mono_path, LPWSTR msi_path)
         TRACE("found support msi version %s at %s\n", versionstringbuf, debugstr_w(msi_path));
 
-        if (compare_versions(WINE_MONO_VERSION, versionstringbuf) <= 0)
+        if (compare_versions(WINE_MONO_VERSION, versionstringbuf) == 0)
         {
             ret = TRUE;
@@ -830,5 +830,5 @@ static BOOL install_wine_mono(void)
         TRACE("found installed support package %s\n", versionstringbuf);
 
-        if (compare_versions(WINE_MONO_VERSION, versionstringbuf) <= 0)
+        if (compare_versions(WINE_MONO_VERSION, versionstringbuf) == 0)
         {
             TRACE("support package is at least %s, quitting\n", WINE_MONO_VERSION);
--
